daemon/cmd/endpoint.go

```diff
@@ -178,15 +178,25 @@ func getEndpointIDHandler(d *Daemon, params GetEndpointIDParams) middleware.Resp
 // The returned pod is deepcopied which means the its fields can be written
 // into.
 func (d *Daemon) fetchK8sMetadataForEndpoint(nsName, podName string) (*slim_corev1.Pod, *endpoint.K8sMetadata, error) {
-	ns, p, err := d.endpointMetadataFetcher.Fetch(nsName, podName)
+	p, err := d.endpointMetadataFetcher.FetchPod(nsName, podName)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	metadata, err := d.fetchK8sMetadataForEndpointFromPod(p)
+	return p, metadata, err
+}
+
+func (d *Daemon) fetchK8sMetadataForEndpointFromPod(p *slim_corev1.Pod) (*endpoint.K8sMetadata, error) {
```

Comment:
stylistic nit: what if this function took the full namespace? Then it no longer needs to return `error`.

Reply:
Aha, I see it's referenced below. Feel free to disregard.

Reply:
Yep, the reason is that it is now also directly called by `createEndpoint` after retrieving the pod from k8s, where we don't have the namespace object already available.